### PR TITLE
add verbosity to pg connections for the backend app

### PR DESCRIPTION
**Pull request by bcollard**
Signed-off-by: Baptiste Collard <9626829+bcollard@users.noreply.github.com>

Add DNS lookup for the PostgreSQL DB and improve logging for the backend nodejs app.

### CODE REVIEW

File: .github/workflows/docker-build-push.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+on:
+  push
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
```
Comment (slonka) [inline .github/workflows/docker-build-push.yaml:12]: we're locking  to specific commits in other repos, please do that as well here:

```suggestion
        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
```

File: .github/workflows/docker-build-push.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      - name: Build and push
+        uses: docker/build-push-action@v5
+        with:
+          file: ./api/Dockerfile
+          context: ./api
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: ${{ secrets.DOCKERHUB_USERNAME }}/kuma-demo-be:latest
+
+
```
Comment (slonka) [inline .github/workflows/docker-build-push.yaml:35]:
```suggestion

```

File: api/app/postgresql.js [hunk 3]
```diff
@@ -1,23 +1,37 @@
 const items = require("../db/items.json");
 const { Pool } = require("pg");
+const pino = require('pino');
+const logger = pino({ name: 'kuma-backend-pg' });
+const dns = require('dns');
+dnsPromises = dns.promises;
```
Comment (slonka) [inline api/app/postgresql.js:6]: Is `dnsPromises` declared anyway?

```suggestion
const dnsPromises = dns.promises;
```

File: api/app/postgresql.js [hunk 4]
```diff
@@ -42,14 +56,14 @@ const importData = () => {
       });
       await client.query("COMMIT");
     } catch (e) {
-      console.log("Error");
+      logger.error('search error: ', err);
       await client.query("ROLLBACK");
       throw e;
     } finally {
-      console.log("Release");
+      logger.error("Release");
```
Comment (slonka) [inline api/app/postgresql.js:63]: why is this an error? 

```suggestion
      logger.info("Release");
```

File: api/index.js [hunk 5]
```diff
@@ -39,17 +42,30 @@ app.post("/upload", async (req, res) => {
...
   postgresql
     .search(req.query.q)
     .then(async (results) => {
+      if (results === undefined) {
+        logger.error('result is undefined');
+      }
+      if (results.rows === undefined) {
+        logger.error('result.rows is undefined');
+      }
+      // logger.info('result: ' + JSON.stringify(results));
```
Comment (slonka) [inline api/index.js:55]: leftover from debugging? either uncomment or remove

**Review by slonka — CHANGES_REQUESTED** [review]
couple of things to change

**Comment by bcollard** [comment]
Thanks for the review.
All suggestions were applied.

**Review by slonka — APPROVED** [review]

**Comment by slonka** [comment]
@bcollard - is this PR ready to be merged?

**Comment by bcollard** [comment]
Hey! Sorry about the delay! 
Yes, it's ready for another round of review.

Summary:
- pinned the version of docker base images
- log DNS details about the Postgres Host
- switched to full async CRUD methods 
- adopted PINO as the underlying logging system
- use GHA to create AMD and ARM docker images
- use GHA to push images to a docker hub account that should be configured with GHA Secrets:
```yaml
      - name: Login to Docker Hub
        uses: docker/login-action@v3
        with:
          username: ${{ secrets.DOCKERHUB_USERNAME }}
          password: ${{ secrets.DOCKERHUB_TOKEN }}
```

File: .github/workflows/docker-build-push.yaml [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+          tags: ${{ secrets.DOCKERHUB_USERNAME }}/kuma-demo-be:latest
+
+      - name: Build and push frontend
+        uses: docker/build-push-action@v5
+        with:
+          file: ./app/Dockerfile
+          context: ./app
+          platforms: linux/amd64,linux/arm64
+          push: true
+          tags: ${{ secrets.DOCKERHUB_USERNAME }}/kuma-demo-fe:latest
```
Comment (lahabana) [inline .github/workflows/docker-build-push.yaml:42]:
```suggestion
          tags: ${{ vars.DOCKER_REPOSITORY || vars.DOCKERHUB_USERNAME }}/kuma-demo-fe:latest
```

**Review by lahabana — COMMENTED** [review]
Thx!

**Review by jakubdyszkiewicz — APPROVED** [review]

**Pull request merged**